### DISCUSSION TITLE
How to generate XLS\XLSX reports from BIRT which can be used without showing security error popup in Microsoft Excel.

### DISCUSSION THREAD

**Discussion by devamitkochar** [eclipse-birt/birt#1183] [Q&A]
We are using BIRT version 2.5.1 , with some jars like viewservlets at 4.4.2 . We need to generate XLS\XLSX reports which do not have XML formatted content so that when customer opens the report, no security popup as given below is shown.

![image](https://user-images.githubusercontent.com/94097869/213163053-ff843ac3-f2fb-4832-adea-0b298dc2ef2e.png)

**Comment by wimjongman** [comment]
What exactly is your question?

**Reply by wimjongman** [reply]
Sorry, I see it in the subject. What happens if you open the report? Is it correct?

We do not support BIRT version 2.5.1. Please update to the latest version and see if this resolves your issue.

**Comment by hvbtup** [comment]
I remember that error message in a different context: When creating a *.doc file with the old Word emitter (DOC).
The Word (DOC) and Excel (XLS) emitters do not create the *binary* Office 97 files that the file suffixes *.doc and *.xls imply.
Instead, they create XML in Office 2003 format.
But if one uses an *.xml file suffix, nobody knows what kind of file that would be, so one has to use a suffix which makes is clear which application is responsible for opening the file.
The warning is because the actual file format and the format implied by the file extension do not match.
Despite the warning, Excel resp. Word will be able to show the content.

Advice:
* Use the latest BIRT version.
* Use the XLSX emitter to create Excel files - it will create an *.xlsx file (and the content is XLSX format)
* For Word output, use the DOCX emitter.
* Try to avoid the old Office 97 binary formats *generally*. They are causing all kinds of security issues. Instead, use the "new" ZIP-bases formats *.docx and *.xlsx

**Reply by wimjongman** [reply]
Thanks, Henning!

**Reply by devamitkochar** [reply]
Which minimum version of BIRT supports export to XLSX , with JAVA 8.0 ? From where we can get the required dependency jars and what version of viewerservlets.jar is required ?

**Reply by hvbtup** [reply]
With Java 8 and XLSX, you can try BIRT 4.4.0 (but note how old that is!) - see https://github.com/eclipse/birt/releases.

Maybe t's time to start using Java 11 in your application... (and yes, I know this is causing a lot of work).

So why don't you download and use BIRT from https://github.com/eclipse/birt/releases/tag/4.10.0, which redirects to https://archive.eclipse.org/birt/downloads/drops/R-R1-4.10.0-202210031241/ ?

**Reply by devamitkochar** [reply]
Thanks Henning, I was trying to use BIRT 4.10.0 but was facing two issues 
1. It needs JDK11 to build and our product currently uses JAVA 8.
2. As a POC I tried to upgrade BIRT to 4.10.0 and using JDK 11, But I was not able to get all dependencies and build kept failing . I am also not aware of latest viewerservlets.jar version to support BIRT 4.10.0. 

Is there any location where I can get all dependencies for BIRT 4.10.0 and Sample project to verify if XLSX export would work with my setup?

**Reply by hvbtup** [reply]
The easiest way to test BIRT ist to use the `report-designer-all-in-one`. Just download, unzip, start. 
I don't know anything about the viewerservlets.jar you mention.
If you download the `birt-runtime`: The ZIP contains a shell script to test running a single report, as well as `birt.war`, which contains everything you need to make a BIRT viewer as part of a web server (I never used it personally, though).

**Reply by devamitkochar** [reply]
Thanks Henning. I would try as you have suggested.

**Reply by devamitkochar** [reply]
Hi Henning, I have updated my project to use BIRT 4.8.0 and I am able to generate XLSX reports . But after BIRT upgrade to 4.8.0 we are observing security vulnerabilities in org.eclipse.birt.report.data.oda.excel_4.8.0.v201806261756.jar and uk.co.spudsoft.birt.emitters.excel_4.8.0.v201806261756 plugin. Can you please suggest alternative BIRT version with no security vulnerabilities

**Reply by hvbtup** [reply]
Still using Java 8 is a security issue as well...

**Reply by merks** [reply]
Problems can only be addressed in a new release so if you are stuck on an old release and not using the latest release, you will be stuck on old problems forever.
